### PR TITLE
Add duplicate scene functionality

### PR DESCRIPTION
**Pull request by emericklaw**
## Proposed change

This change add the ability to duplicate a scene. The other similar sections (automations and scripts) already had this.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

None required, just need to be able to use scenes.

## Additional information

- This PR is related to issue or discussion:
https://community.home-assistant.io/t/scenes-duplicate/299946
https://community.home-assistant.io/t/duplicate-automations-scripts-scenes-for-faster-building-ui/197981

- Link to documentation pull request:
https://github.com/home-assistant/home-assistant.io/pull/17809

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [X] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @emericklaw, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;frontend#9175).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: src/panels/config/scene/ha-scene-editor.ts [hunk 1]
```diff
@@ -683,6 +735,31 @@ export class HaSceneEditor extends SubscribeMixin(
...
+          text: this.hass!.localize(
+            "ui.panel.config.scene.editor.unsaved_confirm"
+          ),
+          confirmText: this.hass!.localize("ui.common.leave"),
+          dismissText: this.hass!.localize("ui.common.stay"),
+        }))
+      ) {
+        return;
+      }
+      // Wait for dialog to complate closing
```
Comment (bramkragten) [inline src/panels/config/scene/ha-scene-editor.ts:751]:
```suggestion
      // Wait for dialog to complete closing
```

File: src/panels/config/scene/ha-scene-editor.ts [hunk 2]
```diff
@@ -83,7 +91,7 @@ export class HaSceneEditor extends SubscribeMixin(
 
   @property() public route!: Route;
 
-  @property() public sceneId?: string;
+  @property() public sceneId!: string;
```
Comment (bramkragten) [inline src/panels/config/scene/ha-scene-editor.ts:94]: Why was this changed?

**Review by bramkragten — APPROVED** [review]
Looks good thanks! One question/remark.

File: src/panels/config/scene/ha-scene-editor.ts (hunk 1 above)
Comment (emericklaw) [reply src/panels/config/scene/ha-scene-editor.ts]: Fixed spelling mistake

File: src/panels/config/scene/ha-scene-editor.ts [hunk 3]
```diff
@@ -198,19 +206,52 @@ export class HaSceneEditor extends SubscribeMixin(
...
+              .path=${mdiContentDuplicate}
+            ></ha-svg-icon>
+          </mwc-list-item>
+
+          <mwc-list-item
+            .disabled=${!this.sceneId}
+            aria-label=${this.hass.localize(
+              "ui.panel.config.scene.picker.delete_scene"
+            )}
+            class=${classMap({ warning: this.sceneId })}
```
Comment (emericklaw) [inline src/panels/config/scene/ha-scene-editor.ts:243]:
```suggestion
            class=${classMap({ warning: this.sceneId! })}
```

File: src/panels/config/scene/ha-scene-editor.ts [hunk 4]
```diff
@@ -198,19 +206,52 @@ export class HaSceneEditor extends SubscribeMixin(
...
+            graphic="icon"
+          >
+            ${this.hass.localize("ui.panel.config.scene.picker.delete_scene")}
+            <ha-svg-icon
+              class=${classMap({ warning: this.sceneId })}
```
Comment (emericklaw) [inline src/panels/config/scene/ha-scene-editor.ts:248]:
```suggestion
            class=${classMap({ warning: this.sceneId! })}
```

Comment (emericklaw) [reply src/panels/config/scene/ha-scene-editor.ts:248]: >

File: src/panels/config/scene/ha-scene-editor.ts (hunk 2 above)
Comment (emericklaw) [reply src/panels/config/scene/ha-scene-editor.ts:94]: This could be because I do not know enough about TypeScript.

I was getting the below warnings.
image

Since duplication had already been implimented on the Scripts and Automations pages I looked at how it was achieved and replicated the top-right menu for Scenes.

### Automation Editor
https://github.com/home-assistant/frontend/blob/fe2fe7468f6e7b975e215646b6c4d5eec20dcd92/src/panels/config/automation/ha-automation-editor.ts#L78

https://github.com/home-assistant/frontend/blob/fe2fe7468f6e7b975e215646b6c4d5eec20dcd92/src/panels/config/automation/ha-automation-editor.ts#L176-L193

### Script Editor
https://github.com/home-assistant/frontend/blob/fe2fe7468f6e7b975e215646b6c4d5eec20dcd92/src/panels/config/script/ha-script-editor.ts#L64

https://github.com/home-assistant/frontend/blob/fe2fe7468f6e7b975e215646b6c4d5eec20dcd92/src/panels/config/script/ha-script-editor.ts#L159-L174


### Possible Alternative
Change back to optional
```suggestion
  @property() public sceneId?: string;
```

And use ! on the sceneId in the classMap:
https://github.com/home-assistant/frontend/pull/9175#discussion_r632457549
https://github.com/home-assistant/frontend/pull/9175#discussion_r632457625

Comment (bramkragten) [reply src/panels/config/scene/ha-scene-editor.ts:94]: Cast `sceneId` to a boolean: `Boolean(this.sceneId)`

Comment (bramkragten) [reply src/panels/config/scene/ha-scene-editor.ts:94]: Corrected for script and automation editor here: <https://github.com/home-assistant/frontend/pull/9184>

Comment (emericklaw) [reply src/panels/config/scene/ha-scene-editor.ts]: Thanks, made the changes.

**Review by bramkragten — APPROVED** [review]
